GEODE-7581: Run SSL Benchmarks with JDK11

nonbinaryprogrammer opened the pull request:
Running SSL benchmarks with JDK12 improves their stability. This change
only supports using JDK11 as an alternative JDK since each different JDK
option that Benchmarks supports would have to be installed individually
on the cluster.

* Add flag to SSL Benchmark jinja variables for using JDK11
* Install JDK11 on the cluster when testJVM option with path to JDK11 is
provided.

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

kirklund committed:
GEODE-7256: Cleanup Connection classes and tests

* Minor cleanup of classes and tests

kirklund committed:
GEODE-7256: Cleanup IgnoredException

Minor cleanup of IgnoredException class.

kirklund committed:
GEODE-7256: Cleanup AlterRuntimeCommandDUnitTest

* Mostly just try to improve readability
* Use LogWriterLevel enum

kirklund committed:
GEODE-7256: Rename AlterRuntimeCommandDistributedTest

From AlterRuntimeCommandDUnitTest.

jhuynh1 committed:
Revert "Revert "Revert "GEODE-6807: cache adviseUpdate and adviseAllEventsOrCached" (#4189)" (#4212)"

This reverts commit 80efe5cae9848f4fab45bb5c199c1e1f1fdf0eae.

mkevo committed:
GEODE-6927 make getThreadOwnedConnection code thread safe (#4085)

* GEODE-6927 make getThreadOwnedConnection code thread safe

* changes due to comments

* fix spotlessAply

* fix clean method

* remove null checks

* empty commit to re-trigger CI

gemzdude committed:
GEODE-7567: Improve cleanup logic in create_instance.sh (#4459)

* GEODE-7567: Improve cleanup logic in create_instance.sh

* GEODE-7567: Increase number of retry attempts

demery-pivotal committed:
GEODE-7568: Pass persistence service to config mgr constructors (#4462)

* GEODE-7568: Pass persistence service to config mgr constructors

Pass the persistence service to each manager on construction, rather
than passing it to each method.

Co-authored-by: Dale Emery <demery@pivotal.io>

* Remove spurious commented code

jinmeiliao committed:
GEODE-7398: add timeDeployed to the runtime info of deployed jar (#4460)

Co-authored-by: Dale Emery <demery@pivotal.io>

jhuynh1 committed:
Revert "Revert "Revert "GEODE-6807: cache adviseUpdate and adviseAllEventsOrCached" (#4189)" (#4212)" (#4442)

This reverts commit 80efe5cae9848f4fab45bb5c199c1e1f1fdf0eae.

kirklund committed:
GEODE-7503: Cleanup GemFireCacheImpl (#4436)

* Move inner-classes to bottom of outer class

* Move fields to top of class

* More fields

* Fixup

* Fix IJ inspection warnings

* Remove bug system ticket numbers

* Cleanup javadocs, toString, and comments

* Use @VisibleForTesting annotation

* Move toString method and backupFiles field

* Pull javadocs up to interfaces.
Define VisibleForTesting methods in InternalCacheForTesting.
Identify more methods as VisibleForTesting.

* Fixup javadocs and minor warnings in Cache interfaces.

* Move unchecked util methods to UncheckedUtils inner class

* Move static constants and fields to top of class.
Rename FIVE_HOURS to FIVE_HOURS_MILLIS.
Review older TODOs.

* Move volatile fields above non-final fields

* Move non-final fields below volatile and final fields

* Move final fields above volatile fields

* wip

* Adjust whitespace and line feeds.
Remove more TODOs and comments.
Fix spelling typo.

* Import various static methods and constants
Adjust more line feeds
Adjust more white space

* Weaken declared types where it makes sense
Add more static imports

* Remove unnecessary uses of final on local vars

* Remove uncessary uses of final on local vars

* Fixup unchecked types
Restore return types to match the interfaces

* Minor cleanup

* Change Consumer<Void> to Runnable (duh)

* Adjust line feeds of large if-statement

* Remove SuppressWarnings

* Fix AssemblyContentsIntegrationTest

* Fixup GemFireCacheImpl

mhansonp committed:
Mass test script change (#4456)

* Improvements to mass test run script

* found a issue with the count of runs

Commit:
GEODE-7537: Remove the synchronization in CacheFactoryStatics.getAnyInstance. (#4435)

jinmeiliao committed:
GEODE-7570: close cache after test is done to prevent polluting other tests (#4463)

Bill committed:
GEODE-7397: create new module: geode-tcp-server (#4450)

jujoramos committed:
GEODE-7535: Prevent exception throw during getAll (#4415)

Added a try-catch block during basicGetAll to avoid throwing
EntryDestroyedException whenever the value is locally destroyed as a result
of another concurrent region operation (like eviction/expiration).
The  operation will try to retrieve the value from the server as it
regularly does for any entry that doesn't exist locally.

- Fixed minor warnings.
- Added unit and distributed tests.

kirklund committed:
GEODE-7256: Move CacheFactoryStaticsTest to IntegrationTest

RemoteGfManagerAgentTest and CacheFactoryStaticsTest failed in my PR
precheckin, so I studied the tests and related changes that were
recently merged in.

Move CacheFactoryStaticsTest to IntegrationTest. Rename to
CacheFactoryStaticsIntegrationTest. This test is creating a full Cache
which makes it an IntegrationTest.

Remove unnecessary spy from RemoteGfManagerAgentTest.

Even though RemoteGfManagerAgentTest was setting the system field in
RemoteGfManagerAgent, the test intermittently nulls out the system and
then creates a real connection to the DistributedSystem which is not
desirable in a unit test, so I replaced static dependencies with
factories that can be stubbed to inject the desired mocks.

Cleanup RemoteGfManagerAgent, RemoteGfManagerAgentTest, and
CacheFactoryStaticsIntegrationTest. Fix up unchecked types and make all
fields private again. Add VisibleForTesting annotation to applicable
methods. Use constructor injection to avoid breaking encapsulation.

Bill committed:
GEODE-7569: repair play dead in membership tests (#4467)

mhansonp committed:
fixing use of concourse_job_performance.py (#4469)

mhansonp committed:
moving a quote to the right place (#4470)

Subverting the not-finished jobs, because this script doesn't interact with them.

mhansonp committed:
Yet another tweak to the jinja template (#4471)

-- Because the upstream concept doesn't seem necessary

jmelchio committed:
Geode 7424 create index (#4458)

co-authored-by: Jinmei Liao <jliao@pivotal.io>

mhansonp committed:
Removing a dependency on my home directory (#4473)

Bill committed:
GEODE-7563: eliminate home-grown ConcurrentHashSet (#4468)

jmelchio committed:
GEODE-7574: fix javadoc warnings (#4472)

Authored-by: Joris Melchior <joris.melchior@gmail.com>

robbadler committed:
Use "gci-resource" to trigger google compute image rebuilds. (#4451)

Rebuild our heavy-lifter images when the upstream base image has been
updated by the platform owner.

jinmeiliao committed:
GEODE-7424: create index (#4474)

* revert accidental change to the logging

Bill committed:
GEODE-7397:  remove spurious archunit clauses (#4480)

nonbinaryprogrammer committed:
GEODE-7581: Run SSL Benchmarks with JDK11

Running SSL benchmarks with JDK12 improves their stability. This change
only supports using JDK11 as an alternative JDK since each different JDK
option that Benchmarks supports would have to be installed individually
on the cluster.

* Add flag to SSL Benchmark jinja variables for using JDK11
* Use jdk11 baked into the image

jake-at-work commented on ci/scripts/run_benchmarks.sh:
```diff
@@ -75,6 +75,10 @@ do
     METADATA_BASELINE="'benchmark_version':'${BASELINE_VERSION}'"
   fi
 
+  if [[ ${FLAGS} == *"-PtestJVM=/usr/lib/jvm/java-11-openjdk-amd64/"* ]]; then
```
I worry too many moving parts gives us more places to fail. How about we bake Java 11 into the base image similar to our test images.

See https://github.com/apache/geode-benchmarks/blob/develop/infrastructure/scripts/aws/image/packer.json

nonbinaryprogrammer committed:
GEODE-7554: Add retry mechanism for failed tests (#4461)

* GEODE-7554: Add retry mechanism for failed tests

There are still some flaky benchmarks. Retry tests that have failed, up
to 5 times, to determine if the failure is legitimate or just a flaky
test. If the test fails 5 times in a row, we know that it is a
legitimate failure.

With each run of run_against_baseline (in the apache/geode-benchmarks
repo), the class names of all failed tests will be written to a file.
The run_benchmarks script called by CI will read the failed tests from
the file and run only those tests. Once the return code is 0, or once
we've tried 5 times, exit.

* build failedFile path dynamically

Signed-off-by: Helena A. Bales <hbales@pivotal.io>

* add print of iteration number for CI debugging of failures

Signed-off-by: Robert Houghton <rhoughton@pivotal.io>

* triggering CI again. Previous status never updated

Signed-off-by: Helena A. Bales <hbales@pivotal.io>

* refactor if statement for readability

* restarting CI

* remove hardcoded cluster tag

BalaKaza committed:
GEODE-7575: Add concourse team specific fly targets in the deploy scripts

Update all the deploy scripts to use team specific fly targets, so that
we do not accidentally deploy to wrong teams.

moleske committed:
Add LGTM Badges (#4406)

- Makes it easier to get to alerts for people to then contribute to Geode

Authored-by: Michael Oleske <moleske@pivotal.io>

Commit:
Revert "GEODE-7537: Remove the synchronization in CacheFactoryStatics.getAnyInstance. (#4435)" (#4482)

This reverts commit 3a3db82afdf0668e98ca49bcbdfa478e0ac94e10.

thefire committed:
GEODE-7533: System Properties catalina.home and catalina.base are cleared during webapp reloads when using HTTP Session Management Module (#4414)

kirklund committed:
GEODE-7525: Prevent NPE in MBeanProxyFactory (#4478)

FederatingManager should not invoke MBeanProxyFactory.removeAllProxies
when monitoringRegion is null.

kirklund committed:
GEODE-7577: Log when waiting for missing persistent members (#4476)

* Separate StartupStatusListenerRegistry from StartupStatus to allow
  non-static injection of StartupStatus into classes that use it
* Add CollectionTransformer interface to TransformUtils to facilitate
  unit testing of classes depending on it
* Add new MembershipChangeListenerFactory with validation to ensure that
  warningDelay and pollDuration have correct values in relation to each
  other (to prevent recurrences of GEODE-7577)
* Fixed the formatting of missing persistent members messages in
  PersistenceAdvisorImpl
* Fixed warnings and minor issues primarily for readability and
  testability in every class touched
* Added shouldSyncForCrashedMember to InternalRegion and pulled the
  javadocs up to the interface

GEODE-4267: Fix PersistentRecoveryOrderOldConfigDUnitTest
* Replace thread sleeps with CountDownLatch
* Add a couple await calls
* Actual fix involved preventing late-running asynchronous actions that
  resulted in logging DistributedSystemDisconnectedException AFTER the
  test already removed the IgnoredException
* Major cleanup of PersistentRecoveryOrderDUnitTest including replace
  usage of Admin API with usage of DistributedSystemMXBean
* Replace Region close with Cache close in test that needs to list and
  revoke missing disk-stores

albertogpz committed:
GEODE-7534: Add example for query with bind params (#4452)

mhansonp committed:
Geode 6374 - Fixing DistributedNoAckRegionCCEOffHeapDUnitTest.testNoLoaderWithInvalidEntry (#4409)

- Changes to wait and single sourcing code.
- Cleaning out AttributeFactory usages
- Cleanup of RegionFactory usages
- Deleting an unneeded test

Commit:
GEODE-7537: use dm.getCache instead of CacheFactory.getAnyInstance (#4486)

kohlmu-pivotal committed:
GEODE-7531: Added type checking before casting generic `Pool.java` to `PoolImpl.java` (#4490)

kohlmu-pivotal committed:
GEODE-7159: Amending emergencyClose to type check before casting. Also included Optional instead of null checks (#4493)

kirklund committed:
GEODE-7586: Fix NPE in RemoteGfManagerAgentTest (#4492)

If the thread invoking disconnect gets enough cpu cycles before the
thread invoking removeMember, then it tries to invoke listApplications
with null for system.

The fix sets the system so the field is non-null which also simulates a
real run more closely as well.

dschneider-pivotal committed:
GEODE-7424: fix acceptance failures (#4484)

Co-authored-by: Joris Melchior <joris.melchior@gmail.com>
Co-authored-by: Darrel Schneider <dschneider@pivotal.io>

onichols-pivotal committed:
add missing variable (#4497)

Commit:
GEODE-7588: Add geode-all-bom to geode-gfsh. (#4491)

robbadler committed:
GEODE-7590 - More Packer restarts to fix Windows image. (#4495)

Trying to have Packer be more careful (lots of restarts) when creating
the Windows heavy-lifter image.
* Move from Packer v1.3.5 to latest, v1.4.5
* Drop unused openjdk:8 docker image from Windows compute image.
* create file before set-content in powershell
* Force file-create in PowerShell

Co-authored-by: Robert Houghton <rhoughton@pivotal.io>
Co-authored-by: Scott Jewell <sjewell@pivotal.io>

jake-at-work committed:
Fixes bash error in run_benchmarks.sh script. (#4498)

robbadler committed:
GEODE-7596 - Use ASF repo:status token sparingly. (#4501)

For polling GitHub for PRs, don't waste ASF rate-limitted API calls.
* Use apachegeode-ci account for all queries.
* Use asf-ci account for all status-sets.

Co-authored-by: Robert Houghton <rhoughton@pivotal.io>
Co-authored-by: Scott Jewell <sjewell@pivotal.io>

dependabot[bot] committed:
Bump excon from 0.65.0 to 0.71.0 in /geode-book (#4481)

Bumps [excon](https://github.com/excon/excon) from 0.65.0 to 0.71.0.
- [Release notes](https://github.com/excon/excon/releases)
- [Changelog](https://github.com/excon/excon/blob/master/changelog.txt)
- [Commits](https://github.com/excon/excon/compare/v0.65.0...v0.71.0)

Signed-off-by: dependabot[bot] <support@github.com>

jhuynh1 committed:
GEODE-7593: Force index to use Strings instead of PdxStrings when eviction is enabled on region (#4500)

robbadler committed:
GEODE-7599 - Use Google Compute Image resource in Concourse. (#4504)

* Parameterize gcp image-family-name.
* Change geode-build template to set image-family-name.
* Remove reduntant 'do' in pipeline definition.
* Add image-family-name for pr and examples pipelines.
* Remove intermediatary files when deploying meta pipeline.
* Test jobs track when new compute image is used.
* Use gci-resource correctly per platform image family

Authored-by: Robert Houghton <rhoughton@pivotal.io>

jhuynh1 committed:
GEODE-7571: Fix logic determining when query will wait for a repo (#4466)

bschuchardt committed:
GEODE-7556: remove membership dependencies on geode-core exeptions (#4502)

* GEODE-7556: remove membership dependencies on geode-core exeptions

Removed use of geode-core exceptions from membership. DistributionImpl now converts membership exceptions into geode-core exceptions where necessary.

Except for MembershipClosedException the new membership exceptions are all checked exceptions. This let me isolate where the exceptions are used and ensure that they're changed into appropriate geode-core exceptions.

MemberShunnedException is now in the membership module instead of the TcpConduit module. It, too, is a checked exception.

CancelCriterion is also removed from use in the membership module. The Stopper in Services.java doesn't need to be a CancelCriterion to function properly.

Several tests had to be modified to handle our stress-test environement.

* use checkCancelled method

jinmeiliao committed:
GEODE-7583: rule improvement for ssl support (#4499)

mkevo committed:
GEODE-7561: GW sender allow single dispatcher thread (#4441)

* GEODE-7561: GW sender allow single dispatcher thread

* empty commit to re-launch CI

* add test

* empty commit to re-trigger CI

* add distributedTest to verify change

* add test to verify null or invalid order policy

* small update

jinmeiliao committed:
GEODE-7398 :show deployer and deploy time (#4509)

Co-authored-by: Darrel Schneider <dschneider@pivotal.io>
Co-authored-by: Dale Emery <demery@pivotal.io>
Co-authored-by: Jinmei Liao <jiliao@pivotal.io>

bschuchardt committed:
GEODE-7544 Break dependency on ClassPathLoader (#4508)

Look for the jgroups configuration file using regular class loaders
instead of geode-core's ClassPathLoader.

bschuchardt committed:
GEODE-7606 Break dependency on InternalDataSerializer (#4507)

Inject a serializer and deserializer into GMSLocator for use in reading
and writing its persistent membership view.

bschuchardt committed:
ensure that exceptions are translated to those acceptable by geode-core (#4516)

Bill committed:
GEODE-7614: change assert to await in ClientHealthStatsDUnitTest (#4515)

Bill committed:
GEODE-7613: move multicast address init to where it's needed (#4517)

Commit:
GEODE-7541: Removing dependency on SocketCreator/Factory (#4479)

- Pushed static call to SocketCreatorFactory up to DistributionImpl
- SocketCreator is now a parameter to MembershipBuilder
- extracted MembershipSocketCreator interface from SocketCreator
- created adapter layer to handle passthru of SocketCreator calls
- renamed with "Tcp" prefix previously refactored socket creator efforts
- Consolidate socket creator interfaces to just TcpSocketCreator.
- Added checked exception handling to Services and re-arranged the finally block to correctly handle start/started state.

demery-pivotal committed:
GEODE-7592: Simplify startManager() precondition checks (#4510)

Co-authored-by: Dale Emery <demery@pivotal.io>
Co-authored-by: Joris Melchior <joris.melchior@gmail.com>

* LGTM complained about a possible NPE in startManager(). There was no
possibility of an NPE, but precondition-checking code was overly
complex, and difficult for LGTM and humans to analyze.

* Adding the tests required injecting several dependencies.

dschneider-pivotal committed:
GEODE-7425 delete index (#4503)

Co-authored-by: Joris Melchior <joris.melchior@gmail.com>
Co-authored-by: Jinmei Liao <jliao@pivotal.io>
Co-authored-by: Darrel Schneider <dschneider@pivotal.io>

mhansonp committed:
GEODE-7387: SingleHopGetAllPutAllDUnitTest.testRemoveAllInClient is failing (#4511)

* Cleanup
* Minor changes recommended in bug notes

bschuchardt committed:
GEODE-7617: CI failure: GeodeClientClusterManagementSSLTest. getServiceUseClientSSLConfig (#4518)

add an await() so that the cluster configuration service has time to
spin up.

bschuchardt committed:
GEODE-7556e - catch and translate exceptions into geode-core exceptions (#4519)

added a few more translation points.  These changes are for
non-opensource legacy tests that are looking for
ForcedDisconnectException as a cause for a cache disconnecting.  There
is nothing in Geode that expects this so I have not added tests that
require it.

Commit:
GEODE-7541: Clean up of auto-generated catch blocks. (#4520)

* Removed some un-thrown exception clauses.

* spotless

upthewaterspout committed:
GEODE-7545: Remove dependency on AltertingAction from membership

The only place where JGroupsMessenger called AlertingAction was when
sending a message. The only message that could be sent while that thread
local is true is AlertListenerMessage.

Setting isHighPriority to true on AlertListenerMessage instead.
JGroupsMessenger is the only place that flag is checked, so this should
have the same effect as testing the thread local.

bschuchardt committed:
GEODE-7541: revert one change (#4526)

One of the changes for this ticket is causing unit test failures on
some machines - especially Macs - in the form of

java.lang.RuntimeException: Failed to start locator
      Caused by:
      org.apache.geode.GemFireConfigException: This process is attempting to join with a loopback address (gzhou-mbpro/127.0.0.1) using a locator that does not have a local address (/10.255.200.197:60167). On Unix this usually means that /etc/hosts is misconfigured.

jinmeiliao committed:
GEODE-7583: allow "status locator" to work with ssl enabled locators (#4529)

* add "security-property-file" option to the command
* use the ssl properties when connect for status

bschuchardt committed:
GEODE-7541: move SocketCreator localhost lookup to geode-common (#4527)

* GEODE-7541: move SocketCreator localhost lookup to geode-common

General host address lookup remains in SocketCreator while localhost-
related functionality has moved to LocalHostUtil.

* added deprecated SocketCreator.getLocalHost() for use in upgrade tests

* hardening tests for stressTest environment

* merging with develop

* remove use of system properties.  ensure old locator directories are cleaned up

upthewaterspout committed:
GEODE-7627: Moving defaults to MembershipConfig

Moving the defaults of membership related settings to the
MembershipConfig class, and referencing them from DistributionConfig.
This makes it easier to create a Membership with the correct default
values.

Adding default implementations of many of the Membership colaborators,
such as AuthenticatorNoOp and MembershipStatisticsNoOp.

Adding a test of creating a Membership using only membership classes.
This test still has several TODOs related to core classes which were
necessary or helpful to make this work.

Co-authored-by: Bill Burcham <bill.burcham@gmail.com>

upthewaterspout committed:
GEOE-7617: Retry awaits in GeodeClientClusterManagementSSLTest

This test had an awaitility clause. Unfortunately, awaitility only retries the expression if it throws an AssertionError. In this case the clause was throwing an IllegalStateException. We want to retry if the IllegalStateException is thrown. Adding a catch and rethrow to tell awaitility to retry.

jinmeiliao committed:
GEODE-7628: Block jmx mbean creation when no security manager is configured (#4534)

moleske committed:
GEODE-7560: Fix LGTM JTAUtils Issues (#4413)

* Remove unused container
* Remove empty conditional
* Use try with resources
* Rewrite do while loop without a break to help LGTM
* Remove unused functions from internal class
* Remove unneeded variables and deprecated usages
* Cleanup names/spellings

Authored-by: Michael Oleske <moleske@pivotal.io>
